### ISSUE TITLE
Refuse overflow hidden by expit in unseen-cohort composition

### ISSUE THREAD

**Issue by bschilder** [bschilder/genomeOS#200]
## Reproduced problem

At `fa6c3db9fbcffb73544f614a3f24e99a20647bf4`, the new pure observation composer can
accept finite inputs, overflow during effect addition, and return a plausible but wrong
endpoint probability. This violates the fail-closed numerical contract (Atlas design §§7.1,12;
#191). It is an extreme-domain stress case, not evidence that ordinary posterior draws are wrong.

## Reproduction

Run with the modeling worktree on PYTHONPATH and its recorded NumPy2.4.6/SciPy1.18.1 environment:

```python
import numpy as np
from genomeos.surfaces.observation import (
    SurveyQueries, ObservationModelMetadata, compose_unseen_observations,
)
queries = SurveyQueries(("s",), ("new",), ("alternate",), (0.0,), (0.0,))
metadata = ObservationModelMetadata(
    "new_cohort_count_v1", ("reference", "alternate"), ("training",),
    True, True, "binomial",
)
result = compose_unseen_observations(
    np.array([[-9e307]]), draw_ids=((0, 0),), queries=queries, metadata=metadata,
    design_effect_draws=np.array([[-9e307]]),
    cohort_sd_draws=np.array([1.7e308]), nugget_sd_draws=np.array([1.4e308]),
    concentration_draws=None, seed=42,
)
print(result.mean_draws)
```

Actual: `[[0.0]]` and `RuntimeWarning: overflow encountered in add`.
The seeded normals are0.41832996532437755 and1.2544943667397455. Independent Decimal
arithmetic yields combined logit `6.67453054487085535E+307`, which is positive, not negative.
The initial latent-plus-design addition becomes negative infinity; subsequent finite positive
effects cannot recover from it. `expit(-inf)` hides the overflow as a finite zero probability.

## Expected and acceptance

Refuse nonfinite arithmetic intermediates with an explicit numerical-domain error. Do not clip
parameters, silently ignore the warning, or introduce an unreviewed precision fallback. Keep
finite large positive/negative logits producing expit endpoints as currently specified.

- Add this exact regression before the fix.
- Cover overflow in design addition, scale multiplication, and effect accumulation.
- Preserve seeded sharing, identity/order, immutable outputs and ordinary endpoint tests.
- Run focused composition/scorer tests, smoke and repository gates; independent scoped review.

No restricted data or GPU work is involved. This is an in-scope fix before the dependent GP
adapter, not a reason to weaken model convergence or benchmark performance criteria.
